torch/csrc/jit/codegen/onednn/interface.cpp

```diff
@@ -0,0 +1,145 @@
+    RemoveListMutation(g);
+    GRAPH_DUMP("After mutation removal. Before DecomposeOps", g);
+    DecomposeOps(g);
+    GRAPH_DUMP("After DecomposeOps. Before PrepareBinaryForLLGA", g);
+    PrepareBinaryForLLGA(g);
+    GRAPH_DUMP("After PrepareBinaryForLLGA. Before DeferSizeCheck", g);
+    DeferSizeCheck(g);
+    GRAPH_DUMP("After DeferSizeCheck. Before CreateLlgaSubgraphs", g);
+    CreateLlgaSubgraphs(g);
+    GRAPH_DUMP("After CreateLlgaSubgraphs. Before PropagateLayout", g);
+    PropagateLayout(g);
+    GRAPH_DUMP(
+        "After PropagateLayout. Before prepareFusionGroupAndGuardOutputs", g);
+
+    // Add shape guard for profiling mode and wipe the tensor type information
+    // from the IR
+    prepareFusionGroupAndGuardOutputs(g->block());
+    GRAPH_DUMP(
+        "After prepareFusionGroupAndGuardOutputs. Before "
+        "RemoveTensorTypeSpecializations",
+        g);
+    RemoveTensorTypeSpecializations(g);
+    GRAPH_DUMP(
+        "After RemoveTensorTypeSpecializations. End of LLGA optimization pass",
+        g);
+  }
+}
+
+} // namespace onednn
+} // namespace fuser
+
+Operation createLlgaKernel(const Node* node) {
+  auto kernel = std::make_shared<fuser::onednn::LlgaKernel>(node);
+  return [kernel](Stack* stack) {
+    RECORD_FUNCTION(kernel->debugName(), std::vector<c10::IValue>());
+    kernel->run(*stack);
+    return 0;
+  };
+}
+
+RegisterOperators oneDNNFusionGroupOp({
+    torch::jit::Operator(
+        prim::oneDNNFusionGroup,
+        createLlgaKernel,
+        AliasAnalysisKind::INTERNAL_SPECIAL_CASE),
+});
+
+Operation createLlgaGuardKernel(const Node* node) {
+  return [node](Stack* stack) {
+    GRAPH_DEBUG("Guarding node: ", node->kind().toQualString());
+    std::vector<TypePtr> types = node->tys(attr::types);
+    const auto num_inputs = types.size();
+
+    GRAPH_DEBUG("num_inputs to guard: ", num_inputs);
+
+    for (size_t i = 0; i < num_inputs; i++) {
+      GRAPH_DEBUG("checking input ", i);
+      auto& input = peek(stack, i, num_inputs);
+      const c10::TensorTypePtr& guard_tensor_type =
+          types[i]->cast<TensorType>();
+
+      if (!input.isTensor()) {
+        GRAPH_DEBUG("input ", i, " is not a tensor, return false");
+        push(stack, IValue(false));
+        return;
+      }
+      const at::Tensor& tensor = input.toTensor();
+
+      // If input tensor is of mkldnn, it's originated from an upstream
+      // LLGA partition that has passed the check on input shapes.
+      // It is valid to continue here as long as the output shapes from
+      // oneDNN graph partitions are determined by the input shapes.
+      if (tensor.is_mkldnn()) {
+        GRAPH_DEBUG("input ", i, " is_mkldnn, continue");
+        continue;
+      }
+
+      if (!guard_tensor_type->matchTensor(tensor)) {
+        GRAPH_DEBUG("input ", i, " check failed, return false");
+        push(stack, IValue(false));
+        return;
+      }
+    }
+
+    // TODO: check type and return the right flag
```

Comment:
`TODO: check type and return the right flag` what do you mean by this ?

Reply:
Thanks for catching this! This comment is a remnant of a time long past 😄 
I'll remove it.